configfile: "config/config.yaml"

include: "rules/assembly.smk"
include: "rules/annotation.smk"
include: "rules/analysis.smk"

rule all:
    input:
         #fastqc,
         "results/Genomics/1_Assembly/1_Preprocessing/fastqc/",
         ##flye
         "results/Genomics/1_Assembly/2_Assemblers/flye/assembly.fasta",
         #winnowmap evaluation
         expand("results/Genomics/1_Assembly/3_Evaluation/winnowmap/{assembler}/merlyDB",
                assembler=[ "flye"],
                genome=["barkhanus"]),
         expand("results/Genomics/1_Assembly/3_Evaluation/winnowmap/{assembler}/repetitive_k15.txt",
                assembler=[ "flye"],
                genome=["barkhanus"]),
         expand("results/Genomics/1_Assembly/3_Evaluation/winnowmap/{assembler}/{long_reads}.bam",
                assembler=[ "flye"],
                long_reads=["nanopore"]),
         #quast
         expand("results/Genomics/1_Assembly/3_Evaluation/quast/{assembler}/",
                assembler=[ "flye"]),
         #deeptools
         expand("results/Genomics/1_Assembly/3_Evaluation/deeptools/{assembler}.png",
                assembler=[ "flye"]),
         #multiqc
         #expand("results/Genomics/1_Assembly/3_Evaluation/multiqc/{assembler}",
          #       assembler=["flye"]),
         #prodigal
         expand("results/Genomics/2_Annotation/1_Structural/prodigal/{assembler}/genome.gff",
                assembler=["flye"]),
         #diamond_blastp
         expand("results/Genomics/2_Annotation/2_Functional/blastp/{annotation}/{assembler}/{db}/genome.blastp",
                 annotation=["prodigal"],
                  assembler=["flye"],
                  db=["diplo_prot"]),
        #Genome level analysis
        #repeatmodeler
         expand("results/ComparativeGenomics/1_GenomeStructureLevel/RModeler/{assembler}/genome_db-families.fa",
                assembler=["flye"]),
         #repeatmasker
         expand("results/ComparativeGenomics/1_GenomeStructureLevel/RMasker/{assembler}",
                assembler=["flye"]),
         #tRNAscan
         expand("results/ComparativeGenomics/1_GenomeStructureLevel/tRNAscan/{assembler}/genome.tRNAscan",
                assembler=["flye"]),
         #tRNAscan_cov
         expand("results/ComparativeGenomics/1_GenomeStructureLevel/tRNAscan/sensitive_search/{assembler}/genome.tRNAscan_cov",
                assembler=["flye"]),
         #barrnap
         expand("results/ComparativeGenomics/1_GenomeStructureLevel/barrnap/{assembler}/genome.rrna.gff",
                assembler=["flye"]),
         #cdhit
         expand("results/ComparativeGenomics/1_GenomeStructureLevel/cdhit/{assembler}/genome_{n}.cdhit",
                assembler=["flye"],
                n=[0.70, 0.75, 0.80, 0.85, 0.90, 0.95, 1])



